src/approaches/grammar_search_invention_approach.py

```diff
@@ -293,7 +301,7 @@ def enumerate(self) -> Iterator[Tuple[Predicate, float]]:
         feature_ranges = self._get_feature_ranges()
         # 0.5, 0.25, 0.75, 0.125, 0.375, ...
         constant_generator = _halving_constant_generator(0.0, 1.0)
-        for c in constant_generator:
+        for n, c in enumerate(constant_generator):
```

Comment:
What's going on here? Why should 0.25 have a lower cost than 0.75? Maybe let's make halving_constant_generator return reasonable costs instead? E.g. 0.5 is lowest cost, then 0.25 and 0.75 are the next-lowest, etc?